operators/advanced_network/python/tune_system.py

```diff
@@ -0,0 +1,629 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import os
+import re
+import subprocess
+import sys
+
+
+def setup_logging():
+    """
+    Configures the logging settings.
+    """
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+
+def parse_args():
+    """
+    Parses command-line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Check system tuning for ANO performance",
+        epilog=(
+            "Examples:\n"
+            f"  python {sys.argv[0]} --check cpu-freq    # Check CPU frequency governor\n"
+            f"  python {sys.argv[0]} --check mrrs        # Check MRRS settings for NVIDIA NICs\n"
+            f"  python {sys.argv[0]} --check mps         # Check max payload size settings\n"
+            f"  python {sys.argv[0]} --set mrrs          # Set PCIe MRRS\n\n"
+        ),
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    group = parser.add_mutually_exclusive_group(required=True)
+
+    group.add_argument(
+        "--check",
+        choices=[
+            "all",
+            "cpu-freq",
+            "mrrs",
+            "mps",
+            "hugepages",
+            "gpu-clocks",
+            "bar1-size",
+            "topo",
+            "cmdline",
+            "mtu",
+        ],
+        help=(
+            "Specify the property to check:\n"
+            "  all        - Perform all checks\n"
+            "  cpu-freq   - Check if the CPU frequency governor is set to 'performance'.\n"
+            "  mrrs       - Check if the Maximum Read Request Size (MRRS) of NVIDIA NICs is set to 4096.\n"
+            "  mps        - Check if the Maximum Payload Size is set to 256B.\n"
+            "  hugepages  - Check if hugepages are enabled\n"
+            "  gpu-clocks - Check GPU clocks\n"
+            "  bar1-size  - Check the BAR1 size of the GPU\n"
+            "  topo       - Check the GPU and NIC topology\n"
+            "  cmdline    - Check the kernel boot parameters\n"
+            "  mtu        - Check MTU of each NVIDIA interface\n"
+        ),
+    )
+
+    group.add_argument("--set", choices=["mrrs"], help=("  mrrs      - Update MRRS of NICs\n"))
+
+    # Check if no arguments are provided
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit(1)
+
+    return parser.parse_args()
+
+
+def get_nic_info():
+    """
+    Parses the output of `ibdev2netdev -v` to extract and return a list of tuples,
+    where each tuple contains the interface name and its PCIe address.
+
+    Returns:
+        List[Tuple[str, str]]: A list of tuples with (interface_name, pcie_address).
```

Comment:
So that's not correct, looks like it returns `Tuple[List[str])`, specifically two lists. Would have been nicer to do what your doc says though (and I guess what you had meant to do initially), to avoid looping over the `idx`.

Reply:
want to switch to a `List[Tuple[str, str]]` and simplify the for loops, or just update that description for now?

Reply:
Fixed